llvm/lib/Target/AMDGPU/GCNNSAReassign.cpp

```diff
@@ -237,7 +237,7 @@ GCNNSAReassign::CheckNSA(const MachineInstr &MI, bool Fast) const {
 
 bool GCNNSAReassign::runOnMachineFunction(MachineFunction &MF) {
   ST = &MF.getSubtarget<GCNSubtarget>();
-  if (!ST->hasNSAEncoding())
+  if (!ST->hasNSAEncoding() || ST->getGeneration() > GCNSubtarget::GFX11)
```

Comment:
If it doesn't have NSA encodings, why isn't it covered by !ST->hasNSAEncoding()?

Reply:
It _only_ has NSA encoding, so there is no point in running a pass that tries to optimize it back to a non-NSA encoded form (I guess we could add a `hasNonNSAEncoding` hook for that).

Reply:
Better to hide it in the named subtarget predicate

Reply:
Done.